src/cpp/src/group_beam_searcher.cpp

```diff
@@ -366,10 +348,12 @@ void update_position_ids(ov::Tensor&& position_ids, const ov::Tensor&& attention
 namespace ov {
 namespace genai {
 
-EncodedResults beam_search(ov::InferRequest& lm,
+std::pair<EncodedResults, std::vector<int32_t>> beam_search(ov::InferRequest& lm,
                            ov::Tensor input_ids,
                            ov::Tensor attention_mask,
-                           GenerationConfig config) {
+                           GenerationConfig config, 
+                           std::optional<ov::Tensor> position_ids = std::nullopt,
+                           std::optional<std::vector<int32_t>> selected_beam_idx = std::nullopt) {
```

Comment:
do we assume that beam search and chat scenario are compatible?

Reply:
it's possible to use with beam search as well. I couldn't verify it because of the problem of several calls of beam search, but it should work

Reply:
ok, probably, I meant chat scenario with streamer and beam search.
Such scenario does not work, right?

Reply:
beam search fix was merged

Reply:
I have an issue with beam search. Unfortunately it does not works for chat. Firstly, it returned output different that in HF. 

Now, when `selected_beam_idx` is a single int, it fails because of the classical `Eltwise shape infer input shapes dim index: 3 mismatch`. I am trying to root cause the issue

Reply:
fixed that issue